adagucserverEC/CDataReader.cpp

```diff
@@ -1062,6 +1066,11 @@ int CDataReader::open(CDataSource *dataSource, int mode, int x, int y, int *grid
     CDataPostProcessor::getCDPPExecutor()->executeProcessors(dataSource, CDATAPOSTPROCESSOR_RUNBEFOREREADING);
   }
 
+  // For datasets without files, such as the Solar Terminator
+  if (dataSource->getFileName() == NULL || dataSource->getFileName()[0] == '\0') {
```

Comment:
Maybe use a boolean to indicate that this is a dataSource without files? You have the same check twice...